quic/s2n-quic-core/src/buffer/error.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Error<E = core::convert::Infallible> {
+    /// An invalid data range was provided
+    OutOfRange,
+    /// The provided final size was invalid for the buffer's state
+    InvalidFin,
+    /// The provided reader failed
+    ReaderError(E),
+}
+
+impl<E> From<E> for Error<E> {
+    #[inline]
+    fn from(reader: E) -> Self {
+        Self::ReaderError(reader)
+    }
+}
+
+impl Error {
+    /// Maps from an infallible error into a more specific error
+    #[inline]
+    pub fn mapped<E>(error: Error) -> Error<E> {
```

Comment:
What is this doing? Doesn't seem to change the error?

Reply:
Yeah this is when you call a function that returns an `Error<Infallible>` in a function that returns `Error<T>`. The conversion is sound since the reader error can never fail, but we still need to translate the other two enum variants.